onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -984,16 +1046,34 @@ TensorrtExecutionProvider::GetCapability(const GraphViewer& graph,
   const int number_of_ort_nodes = graph.NumberOfNodes();
   std::vector<size_t> nodes_vector(number_of_ort_nodes);
   std::iota(std::begin(nodes_vector), std::end(nodes_vector), 0);
+
   std::vector<size_t> filtered_nodes_vector;
   const std::vector<NodeIndex>& node_index = graph.GetNodesInTopologicalOrder();
-
-  // We currently exclude "If" and "Loop" control flow ops from original node vector before calling TensorRT parser.
-  // The reason is, these control flow ops have subgraph which might contain TRT fused node after ORT partition.
-  // If this is the case, TensorRT parser will complain the non-recognized TRT fused node and fail.
   for (const auto& index : nodes_vector) {
     const auto& node = graph.GetNode(node_index[index]);
-    if (node->OpType() == "If" || node->OpType() == "Loop" || node->OpType() == "Scan") {
-            continue;
+
+    /* If current node is control flow op, we take different approach based on following four cases:
+     * 
+     * (1) control flow op is supported by TRT, and its subgraphs are all supported by TRT. Assign this node to TRT.
+     * (2) control flow op is supported by TRT, but not all its subgraphs supported by TRT. Don't assign this node to TRT.
+     * (3) control flow op is not supported by TRT, but its subgraphs all supported by TRT. Don't assign this node to TRT.
+     * (4) control flow op is not supported by TRT, and not all its subgraphs supported by TRT. Don't assign this node to TRT.
```

Comment:
For cases 2, 3, 4, it's better to clarify that even though the control flow op is not assigned to TRT, any portion of its subgraphs that can run in TRT will be still fused and assigned to TRT EP. Otherwise it will give the false impression that TRT EP can only run in case 1.

Reply:
Sure, will add the description in the comments.